tests/unit/test_signers.py

```diff
@@ -1168,3 +1176,32 @@ def test_custom_region(self):
         self.assertIn(region, result)
         # The hostname won't be changed even if a different region is specified
         self.assertIn(hostname, result)
+
+
+@pytest.mark.parametrize(
+    'request_method, content_type_present',
+    [
+        ('GET', False),
+        ('HEAD', False),
+        ('OPTIONS', False),
+        ('POST', True),
+        ('PUT', True),
+        ('DELETE', True),
+        (None, True),
+    ],
+)
+def test_generate_presigned_url_content_type_removal(
+    polly_client,
+    request_method,
+    content_type_present,
+):
+    url = polly_client.generate_presigned_url(
+        'synthesize_speech',
+        Params={
+            'OutputFormat': 'mp3',
+            'Text': 'Hello world!',
+            'VoiceId': 'Joanna',
+        },
+        HttpMethod=request_method,
+    )
+    assert ('content-type' in url) == content_type_present
```

Comment:
I was going to ask if we could be more specific with this assertion, but it turns out we only add this to the SignedHeaders section and not the query parameters themselves. That means anyone using the URL needs to also have a client to pass the headers. That's likely expected when a body is present, but we should validate this against a couple other SDKs first.

Let's see what Java v2 and JS v3 do in these cases.

Reply:
Here's a `PutObject` example for Java v2. It looks like its expected to manually set headers after generating the presigned URL:
```Java
/ Upload content to the Amazon S3 bucket by using this URL.
            URL url = presignedRequest.url();

            // Create the connection and use it to upload the new object.
            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
            connection.setDoOutput(true);
            connection.setRequestProperty("Content-Type","text/plain");
            connection.setRequestProperty("x-amz-meta-author","Mary Doe");
            connection.setRequestProperty("x-amz-meta-version","1.0.0.0");
```

Reply:
I'm also seeing manually providing headers in the `PutObject` example in this blog post for the JS v3: https://aws.amazon.com/blogs/developer/generate-presigned-url-modular-aws-sdk-javascript/

```javascript
const response = await fetch(presignedUrl, {
    method: "PUT",
    body: payload,
    headers: {
         "Content-Length": statSync(filePath).size
    }
});
```

I can try to look for other examples if you like.

Reply:
I think we're mostly concerned about the content of the string.

Reply:
Sure. Here's a redacted example from JS v3. Note that `content-length` is in `SignedHeaders`:

```zsh
https://bar.s3.us-east-1.amazonaws.com/baz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Content-Sha256=UNSIGNED-PAYLOAD&X-Amz-Credential=xxx%2F20230518%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230518T212834Z&X-Amz-Expires=3600&X-Amz-Security-Token=xxx&X-Amz-Signature=xxx&X-Amz-SignedHeaders=content-length%3Bhost&x-id=PutObject

Reply:
Sorry this took awhile, but here's a redacted Java v2 example.

```zsh
https://foo.s3.amazonaws.com/bar?X-Amz-Security-Token=xxx&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-
Date=20230518T223024Z&X-Amz-SignedHeaders=host&X-Amz-Expires=600&X-Amz-
Credential=xxx%2F20230518%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Signature=xxx